cmd/kubelet/app/server.go

```diff
@@ -425,7 +425,9 @@ func run(s *options.KubeletServer, kubeDeps *kubelet.KubeletDeps) (err error) {
 			if err != nil {
 				return err
 			}
-			if cloud == nil {
+			if strings.EqualFold(s.CloudProvider, cloudprovider.ExternalCloudProvider) {
```

Comment:
This should probably gate the whole block, starting at L423.  Otherwise, InitCloudProvider() will have already been called